chips/cc26x2/src/rfc.rs

```diff
@@ -0,0 +1,580 @@
+            status = command_op.status.get();
+            if status == 0x0400 {
+                return Ok(());
+            }
+
+            timeout += 1;
+        }
+
+        return Err(status as u32);
+    }
+
+    fn cmdsta(&self) -> u32 {
+        let dbell_regs = RFC_DBELL_BASE;
+        let ret: u32 = dbell_regs.cmdsta.get();
+
+        return ret;
+    }
+
+    fn enable_hw_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Enable all interrupts
+        dbell_regs
+            ._rfhwien
+            .modify(RFHWInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    pub fn handle_hw_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Clear all RFHW interrupts
+        dbell_regs
+            ._rfhwifg
+            .write(RFHWInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    fn disable_hw_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Disable all RFHW interrupts
+        dbell_regs
+            ._rfhwien
+            .modify(RFHWInterrupts::ALL_INTERRUPTS::CLEAR);
+        // Clear all RFHW interrupts
+        dbell_regs
+            ._rfhwifg
+            .write(RFHWInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    fn enable_cpe_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Enable CPE interrupts
+        dbell_regs
+            .rfcpeien
+            .modify(CPEInterrupts::ALL_INTERRUPTS::SET);
+    }
+
+    pub fn handle_cpe_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Clear all CPE interrupts
+        dbell_regs.rfcpeifg.set(0x7FFFFFFF);
+    }
+
+    fn disable_cpe_interrupts(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Disable all CPE interrupts
+        dbell_regs
+            .rfcpeien
+            .modify(CPEInterrupts::ALL_INTERRUPTS::CLEAR);
+        // Clear all CPE interrupts
+        dbell_regs.rfcpeifg.set(0x7FFFFFFF);
+    }
+
+    pub fn cpe_vec_select(&self, cpe: bool) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Select CPE0 or CPE1
+        if !cpe {
+            dbell_regs.rfcpeisl.modify(CPEVectorSelect::ALL::CLEAR);
+        } else {
+            dbell_regs.rfcpeisl.modify(CPEVectorSelect::ALL::SET);
+        }
+    }
+
+    pub fn handle_ack_interrupt(&self) {
+        let dbell_regs = RFC_DBELL_BASE;
+        // Reset flag for Command ACK
+        dbell_regs.rfackifg.write(DBellCmdAck::CMDACK::SET);
+    }
+
+    pub fn send<T>(&self, rf_command: &T) -> RfcResult {
+        let command = { (rf_command as *const T) as u32 };
+
+        return self.post_cmdr(command);
+    }
+
+    fn send_direct(&self, dir_command: &cmd::DirectCommand) -> RfcResult {
+        let command = {
+            let cmd = dir_command.command_no as u32;
+            let par = dir_command.command_no as u32;
+            (cmd << 16) | (par & 0xFFFC) | 1
+        };
+
+        return self.post_cmdr(command);
+    }
+
+    pub fn wait<T>(&self, rf_command: &T) -> RfcResult {
```

Comment:
Similar safety issues as with `send`, except technically it seems that the lifetime might be OK in this case (see comment in `wait_cmdr`, though).

(btw, I'm not entirely sure what the argument to this method should be...)

Reply:
The argument here is the same as send, some radio command. They are both being made private for now and only called from the specific command functions. Send is returning a result from the `rfcpeifg` and wait is returning `status` from the command called.